### PR TITLE
Dimension nodes

### PR DESCRIPTION
**Pull request by ZiyunShang**
### Purpose

1. Add Dimension ByElementDirection node.
2. Add icons for some view nodes.
DimensionByElementDirectionNode

#### SystemTests
DimensionByElementDirection

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@AndyDu1985 @wangyangshi @mjkkirschner @QilongTang @Amoursol 

#### ByElementDirection
ByElementDirection_2020-09-21_08-41-36

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Dimension.cs [hunk 1]
```diff
@@ -611,6 +652,44 @@ private static Autodesk.Revit.DB.XYZ GetMidpoint(Autodesk.Revit.DB.BoundingBoxXY
...
+        private static bool IsVertical(Autodesk.Revit.DB.XYZ v1, Autodesk.Revit.DB.XYZ v2)
+        {
+            var m = v1.X * v2.X + v1.Y * v2.Y + v1.Z * v2.Z;
+
+            if (m == 0)
+                return true;
+            else
+                return false;
+        }
+        private const double Tolerance = 0.000001;
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Dimension.cs:670]: I think there maybe other place to have `Tolerance `, could you check about it? And also is there any existing library that has `IsVertical `or `IsParallel`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +15,7 @@
 * Add some View Nodes - View.HideCategoriesTemporary, View.HideElementsTemporary, View.IsolateCategoriesTemporary, View.IsolateElementsTemporary.
 * Improve DuplicateSheet Node - add "duplicateWithContents" option, add default suffix value when prefix & suffix are both empty.
 * Improve DuplicateSheet Node - Set Sheet information when duplicating; Improve View Temporary Nodes.
+* Add 4 UI Nodes - View Duplicate Options, View Disciplines, View DisplayStyles, View PartsVisibilitys
```
Comment (AndyDu1985) [inline CHANGELOG.md:18]: I can't find these 4 UI Nodes in this PR.

Comment (ZiyunShang) [reply CHANGELOG.md:18]: It was forgotten to be recorded in the previous PR, now it is added.

File: src/Libraries/RevitNodes/Elements/Dimension.cs (hunk 1 above)
Comment (ZiyunShang) [reply src/Libraries/RevitNodes/Elements/Dimension.cs]: I just have a new commit, could you help review again?

**Review by AndyDu1985 — APPROVED** [review]

**Review by wangyangshi — APPROVED** [review]

**Pull request merged**